src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs

```diff
@@ -98,7 +98,7 @@ public partial class UpdateDurableFunction
                 await context.CallActivityWithRetryAsync(
                     nameof(DeleteOldVersionBlobAsync),
                     _options.RetryOptions,
-                    instanceWatermarks);
+                    new UpdateInstanceBlobArguments(input.PartitionKey, instanceWatermarks, input.ChangeDataset));
```

Comment:
You cannot change the arguments to an activity/function without versioning the function too. Otherwise it's a breaking change and could break functions that are "in-flight" during an update.

Furthermore, let's use a separate type that way we're not needlessly passing the change dataset too